### ISSUE TITLE
Add Annotations update API

### ISSUE THREAD

**Issue by GoogleCodeExporter** [gzog/analytics-issues#53]
```
--------------------------------------------------------------------------
NOTE: This issue tracking system is for developer products only.  If you
are not a developer/programmer and have a problem with a Google web site,
please report the problem to the appropriate group.  More information can
be found here: http://www.google.com/support/
--------------------------------------------------------------------------

Name of API affected:
Analytics Data Export API

Request summary:
Allow annotations to be added/updated via an API.

Use Cases:
When we release new builds/changes to our sites, we'd utilize the API to
add annotations as part of our release.

```

Original issue reported on code.google.com by `ichc.lo...@gmail.com` on 18 Jan 2010 at 11:50

**Comment by GoogleCodeExporter** [comment]
```
This would also allow us to automate annotations when we send email marketing, 
post to 
blogs or social networks, etc. - would be very powerful!

Issues with current API model though (hint in the name 'Export') as it only 
allows 
access to data not modification.
```

Original comment by `simondwi...@gmail.com` on 22 Jan 2010 at 11:17

**Comment by GoogleCodeExporter** [comment]
```
Issue 1916 has been merged into this issue.
```

Original comment by `nickski1...@gmail.com` on 15 Mar 2010 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
Excellent idea which will definately work for all us marketers.
```

Original comment by `digimarketingconvo@gmail.com` on 25 Mar 2010 at 9:27

**Comment by GoogleCodeExporter** [comment]
```
It would be useful to add this feature so linkbuilding, PR, or ppc change 
activity
can be recorded in GA.

Here are some other examples of useful data to add annotations to...

Google News Events
http://news.google.co.uk/news?um=1&cf=all&ned=uk&hl=en&q=%22my+company+name%22&c
f=all&output=rss

Chart + News on Googlefinance
http://www.google.co.uk/finance?client=ob&q=NASDAQ:GOOG
```

Original comment by `phildpea...@googlemail.com` on 25 Mar 2010 at 6:53

**Comment by GoogleCodeExporter** [comment]
```
We're an ESP and we'd really like to automatically create annotations in our 
clients' GA when they launch a 
mailing.   But another, perhaps better, way to track "external events that 
drive website traffic indirectly" is 
integration with a Google Calendar.
```

Original comment by `ipos...@gmail.com` on 26 Mar 2010 at 7:02

**Comment by GoogleCodeExporter** [comment]
```
This would be useful
```

Original comment by `googlema...@cendyn.com` on 29 Mar 2010 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
Please stop commenting just to say "+1". If you want to follow this issue, you 
can
star it.
```

Original comment by `kaz...@gmail.com` on 29 Mar 2010 at 8:08

**Comment by GoogleCodeExporter** [comment]
```
mechanical
```

Original comment by `maheshahearwal@gmail.com` on 16 Apr 2010 at 1:25

**Comment by GoogleCodeExporter** [comment]
```
This would be an awesome addition and would help our customers to better 
understand
how what we're working on how publishing great content to the web affects their
traffic and repeat visitors.
```

Original comment by `daveconk...@gmail.com` on 3 May 2010 at 4:10

**Comment by GoogleCodeExporter** [comment]
```
ecommerce users could push an annotation when a product goes out and out of 
stock and
use analytics to measure trends in sales.
```

Original comment by `nick.rev...@gmail.com` on 5 May 2010 at 7:49

**Comment by GoogleCodeExporter** [comment]
```
Issue 2270 has been merged into this issue.
```

Original comment by `nickski1...@gmail.com` on 18 Aug 2010 at 7:51

**Comment by GoogleCodeExporter** [comment]
```
With the launch of the Management API, shouldn't this be included? 
http://code.google.com/apis/analytics/docs/mgmt/home.html
```

Original comment by `jmb...@gmail.com` on 1 Sep 2010 at 11:31

**Comment by GoogleCodeExporter** [comment]
```
Anyone know if this is in the new management API?  I have been looking 
feverishly for some sort of information on this subject, but all I seem to be 
able to find is wishful thinking.
```

Original comment by `JohnCoov...@gmail.com` on 8 Sep 2010 at 9:14

**Comment by GoogleCodeExporter** [comment]
```
Hi, John as far as I know the new management won't allow you to add annotations.
```

Original comment by `andres%g...@gtempaccount.com` on 8 Sep 2010 at 9:36

**Comment by GoogleCodeExporter** [comment]
```
wikileaks pakistan ...................................
```

Original comment by `wikileak...@gmail.com` on 2 Dec 2010 at 10:45

**Comment by GoogleCodeExporter** [comment]
```
http://wikileakspakistan.blogspot.com/
```

Original comment by `wikileak...@gmail.com` on 2 Dec 2010 at 10:46

**Comment by GoogleCodeExporter** [comment]
```
http://wikileakspakistan.blogspot.com/
```

Original comment by `wikileak...@gmail.com` on 2 Dec 2010 at 10:47

**Comment by GoogleCodeExporter** [comment]
```
thanks for google
```

Original comment by `demast...@gmail.com` on 17 Dec 2010 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
Issue 2471 has been merged into this issue.
```

Original comment by `nickski1...@gmail.com` on 26 Jan 2011 at 8:34

**Comment by GoogleCodeExporter** [comment]
```
Cant wait to see this :)
```

Original comment by `mrrobertson` on 15 Feb 2011 at 4:21

**Comment by GoogleCodeExporter** [comment]
```
+1 on this request. I'd like to add automation in our website for recording 
when a new post is written as well as number of other things to keep good notes 
on what is going on.
```

Original comment by `melvinr...@gmail.com` on 6 Mar 2011 at 7:23

**Comment by GoogleCodeExporter** [comment]
```
+1, it would make the charts way better... 
```

Original comment by `Sied...@gmail.com` on 17 Mar 2011 at 9:22

**Comment by GoogleCodeExporter** [comment]
```
count me in ... I'd love to see this feature available 
```

Original comment by `brian....@gmail.com` on 30 Mar 2011 at 5:41

**Comment by GoogleCodeExporter** [comment]
```
+1 we want to integrate this into our deploy routine
```

Original comment by `o...@soundcloud.com` on 26 Apr 2011 at 1:36

**Comment by GoogleCodeExporter** [comment]
```
ganesh gangadhar mote mo.7588055738                                             
                           at.salimba                                           
                         tq.wadwani                                             
                            po.pimparkhed dist.beed                             
                                             state.maharashtra 431122   
```

Original comment by `ganesh.m...@gmail.com` on 12 Jun 2011 at 2:39

**Comment by GoogleCodeExporter** [comment]
```
Oh hell yeah. This would be so useful.

If Google does this, I'll write a Wordpress plugin that taps into it. (As 
suggested by Nick here: 
http://webmasters.stackexchange.com/questions/16716/how-could-i-import-twitter-p
osts-as-annotations-in-google-analytics/16719#16719)
```

Original comment by `whaletat...@gmail.com` on 12 Jul 2011 at 4:39

**Comment by GoogleCodeExporter** [comment]
```
Removing an obsolete label that was used when these issues were in the 
gdata-issues project.
```

Original comment by `jrobbins@google.com` on 21 Jul 2011 at 10:04
- Removed labels: **API-Analytics**

**Comment by GoogleCodeExporter** [comment]
```
Changing the issue type to match the organization desired in this project.
```

Original comment by `jrobbins@google.com` on 21 Jul 2011 at 10:06
- Added labels: **Type-Feature**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I hope this gets implemented. This would be extremely useful. 
```

Original comment by `mareks...@gmail.com` on 1 Sep 2011 at 3:14

**Comment by GoogleCodeExporter** [comment]
```
This feature would help put trends in context. 

When considering this feature, it may be helpful to imagine multiple audiences 
using it. For example, as melvin said above, the dev team can automate the 
notifications of code changes. Email marketers can note major newsletter or 
offer mailings. Paid advertisers can note the launch date of campaigns. 
Optimizers can note A/B or MVT test launches, etc. It would be helpful if these 
were categorized so we can see all public dev anotations, all email 
annotations, personal private annotations etc.

Since many of the stakeholders are using various software packages to update or 
report, having API access to read AND write annotations would provide a very 
useful level of functionality. 
```

Original comment by `jmcmil...@intelius.com` on 2 Sep 2011 at 5:05

**Comment by GoogleCodeExporter** [comment]
```
+one on my end

This would allow us to add annotations based on published content for a news 
website I manage. Some kind of csv upload or xml file linking could also be a 
solution.
```

Original comment by `wouter.v...@gmail.com` on 2 Oct 2011 at 12:35

**Comment by GoogleCodeExporter** [comment]
```
This would allow our CMS to auto-insert a comment every time we published 
content - which would be extremely useful.
```

Original comment by `bhough...@financialforce.com` on 10 Nov 2011 at 10:38

**Comment by GoogleCodeExporter** [comment]
```
yes!
```

Original comment by `ctalle...@gmail.com` on 10 Nov 2011 at 10:40

**Comment by GoogleCodeExporter** [comment]
```
I need this feature pls
```

Original comment by `ynu...@gmail.com` on 15 Nov 2011 at 10:10

**Comment by GoogleCodeExporter** [comment]
```
I think this feature would be great as I want to get a hook into Capistrano/git 
so whenever a dev deploys they can give a comment that describes the change 
thats just been implemented. Then looking back we can see what changes actually 
made a difference
```

Original comment by `jamie.wi...@forward.co.uk` on 22 Nov 2011 at 1:02

**Comment by GoogleCodeExporter** [comment]
```
I'm voting for this as well. It would be a huge time saver. In practice it 
would also allow for webmasters/ businesses to track more events that occur and 
therefore make much better decisions for users across the web.  
```

Original comment by `DonnieAllenCooper` on 28 Nov 2011 at 11:33

**Comment by GoogleCodeExporter** [comment]
```
This would be super helpful. This could be don't through webmaster tools 
integration already I would think?
```

Original comment by `optimizationtutor.com` on 30 Nov 2011 at 6:37

**Comment by GoogleCodeExporter** [comment]
```
That would save me a huge amount of time!
```

Original comment by `steve...@rsc.org` on 1 Dec 2011 at 4:42

**Comment by GoogleCodeExporter** [comment]
```
i bet this will just be for gold members.
```

Original comment by `pat.gr...@gmail.com` on 9 Dec 2011 at 4:09

**Comment by GoogleCodeExporter** [comment]
```
Please Please Google AdWords Auto Annotations in Google Analytics...
```

Original comment by `h...@hansakoch.com` on 19 Dec 2011 at 10:14

**Comment by GoogleCodeExporter** [comment]
```
+1, Would be definitely helpful to add this automated step in my deploys 
scripts!
```

Original comment by `dhgar...@gmail.com` on 21 Dec 2011 at 12:57

**Comment by GoogleCodeExporter** [comment]
```
This would be extremely useful
```

Original comment by `martijn....@pon.com` on 27 Dec 2011 at 8:28

**Comment by GoogleCodeExporter** [comment]
```
I need I need I need I need !
```

Original comment by `analyt...@numerique.ca` on 17 Jan 2012 at 8:40

**Comment by GoogleCodeExporter** [comment]
```
This would be really useful. 
```

Original comment by `dob...@gmail.com` on 19 Jan 2012 at 9:54

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Google, what is the word? Any thoughts on when this will be included in the 
product?
```

Original comment by `pat.gr...@gmail.com` on 22 Jan 2012 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
We are really looking forward for this!
```

Original comment by `j...@dagerot.com` on 21 Feb 2012 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
I've been following this issue for nearly two years now. Time to give up hope?
```

Original comment by `badmonkey0001` on 21 Feb 2012 at 10:23

**Comment by GoogleCodeExporter** [comment]
```
@badmonkey0001 I've been wondering the same thing. Is Google even paying 
attention to this?
```

Original comment by `arnold.zokas` on 21 Feb 2012 at 10:49

**Comment by GoogleCodeExporter** [comment]
```
This would be a great addition. I envision being able to push changes to a 
website and use the API to automatically create annotations based off of recent 
SVN/GIT commit messages.
```

Original comment by `jake.e.w...@gmail.com` on 22 Feb 2012 at 10:21

**Comment by GoogleCodeExporter** [comment]
```
Yes, I would use this in conjunction with my website deployment system.
```

Original comment by `jer...@burnbright.net` on 22 Feb 2012 at 10:26

**Comment by GoogleCodeExporter** [comment]
```
guys, this is never gonna happen... let's stop asking!
```

Original comment by `aaoui...@gmail.com` on 22 Feb 2012 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
How can I unsubscribe to this? I'd love to see the feature, but I'm sick of the 
"me too" notifications.  On that note, sorry to everyone for triggering yet 
another email to this request. 
```

Original comment by `simfo...@gmail.com` on 22 Feb 2012 at 11:47

**Comment by GoogleCodeExporter** [comment]
```
No problem; as long as you're fine with getting this one too ;).

Btw; yes - I too still want this feature!

You should be able to unsubscribe by "unstarring" the thread. Though that will 
also undo your vote for this issue. And I guess we all want to keep those 
votes, so that perhaps one day somebody at Google will work on this.

Perhaps you should just make an email filter instead?
```

Original comment by `wouter.v...@gmail.com` on 22 Feb 2012 at 11:56

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
When a ticket has been around long enough to start collecting SPAM, it's a 
definite sign that it's never going to get resolved... 2.5 years now. Geez.
```

Original comment by `badmonkey0001` on 1 Jun 2012 at 6:19

**Comment by GoogleCodeExporter** [comment]
```
Yes yes yes! +1
```

Original comment by `adba...@ualr.edu` on 19 Oct 2012 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
We're increasingly using the Google Analytics API to archive reports and slice 
the data in increasingly complex ways. I would love to see this implemented so 
we can pull our release annotations into our own reporting and have it 
centralized.
```

Original comment by `c.filipe...@gmail.com` on 12 Nov 2012 at 10:31

**Comment by GoogleCodeExporter** [comment]
Original comment by `api.pe...@gmail.com` on 16 Nov 2012 at 3:25
- Added labels: **Component-CoreReportingAPI**
- Removed labels: **Component-DataExportAPI**

**Comment by GoogleCodeExporter** [comment]
```
This API feature is a no brainer. Waiting anxiously!
```

Original comment by `jcorr...@blastam.com` on 20 Dec 2012 at 10:53

**Comment by GoogleCodeExporter** [comment]
```
We'd love to have this, too. 
```

Original comment by `jure...@gmail.com` on 16 Feb 2013 at 2:29

**Comment by GoogleCodeExporter** [comment]
```
Now that Google has made a change to it's User Permissions, this feature is 
invaluable to us as a company as we need the team to be able to add annotations 
that are 'shared' but do not want to give them 'edit' access for security 
reasons. So we are now missing out on valuable information on our trending 
graphs. 

If we had this feature on the API we would be able to do shared annotations 
more securely.

```

Original comment by `shelly.w...@holidayextras.com` on 10 Apr 2013 at 9:37

**Comment by GoogleCodeExporter** [comment]
```
Waste of time. I've been following this for 3 years and Google Still hasn't 
replied of given any idea if this feature is on the roadmap.
```

Original comment by `dara.ho...@abbeytheatre.ie` on 10 Apr 2013 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
I'm still going to promote it. This would be a FANTASTIC feature to have.
```

Original comment by `c.filipe...@gmail.com` on 11 Apr 2013 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
Please at lease support a google doc import/export or mass edit...  Please for 
the love of god let us know you are still thinking about this and wont just 
kill it next week. 
```

Original comment by `t...@brickhousesecurity.com` on 21 Apr 2013 at 3:54

**Comment by GoogleCodeExporter** [comment]
```
Please hurry Google. 
```

Original comment by `michael....@gmail.com` on 11 Jun 2013 at 1:50

**Comment by GoogleCodeExporter** [comment]
```
Hey Google, Help US !!!
```

Original comment by `raf...@rafaelfurquim.com.br` on 2 Jul 2013 at 5:51

**Comment by GoogleCodeExporter** [comment]
```
doit.
```

Original comment by `subs...@gmail.com` on 8 Jul 2013 at 11:18

**Comment by GoogleCodeExporter** [comment]
```
+1 Still waiting on this. I would love to create our own internal annotations 
solution, but without being able to relate it back to our analytics (or being 
able to pull that data into our own system), it doesn't make sense!
```

Original comment by `c.filipe...@gmail.com` on 30 Jul 2013 at 9:28

**Comment by GoogleCodeExporter** [comment]
```
@c.fillipe You can pull the data out of GA, it's surprisingly easy to do.
```

Original comment by `daniel.crompton` on 31 Jul 2013 at 1:07

**Comment by GoogleCodeExporter** [comment]
```
@daniel How do you pull annotations data out of GA?
```

Original comment by `es6.m...@gmail.com` on 31 Jul 2013 at 1:49

**Comment by GoogleCodeExporter** [comment]
```
@c.fillipe sorry I was unclear, you want to create an internal annotation 
solution, so pull all the data out of GA and push it into this internal 
solution. Extracting GA data, with the exception of annotations, is 
surprisingly easy to do.
```

Original comment by `daniel.crompton` on 31 Jul 2013 at 2:12

**Comment by GoogleCodeExporter** [comment]
```
Oh, *much* clearer. LOL.

```

Original comment by `fcher...@gmail.com` on 31 Jul 2013 at 2:18

**Comment by GoogleCodeExporter** [comment]
```
I think the question is "Can I, using my API token, access my annotations data 
programmaticly?" That is to say:

* As an API user, I can send a request to retrieve all of my annotations so 
that I can store them in a third-party system
* As an API user, I can add annotations so that my third-party data can be 
reflected in Google Analytics
* As an API user, I can delete annotations so that my data is up to date

These are essentially the requirements that this ticket is trying to address. 
Additional features could be:

* Retrieve all annotations by date range
* Retrieve all annotations across multiple profiles
```

Original comment by `step...@raventools.com` on 31 Jul 2013 at 2:20

**Comment by GoogleCodeExporter** [comment]
```
My concern at this point is: Google seems to have no interest in this feature, 
in which case: what exactly are annotations meant to be used for? Since, 
apparently, it's not what we're expecting.
```

Original comment by `w...@couturelab.com` on 31 Jul 2013 at 2:34

**Comment by GoogleCodeExporter** [comment]
```
My original request was to be able to ADD annotations to GA via the API.
For example if I have a email program, it could add an annotation to GA every 
time an email is sent to customers, then we can easily look at sales behaviour 
based on that annotation.
```

Original comment by `hogan.dara` on 31 Jul 2013 at 2:54

**Comment by GoogleCodeExporter** [comment]
```
That's not what annotations are for. The correct way of tracking what you
want is with events. Events are really powerful, you should look into it.

Annotations were added to be one-off notes for final users for a date. Like
for example: next time you have a spike in your traffic, you an annotate
the date with a "emails sent to customer", so that when you look at it
again 3 months for now, you can remember what it is.
```

Original comment by `eduardos...@gmail.com` on 31 Jul 2013 at 3:25

**Comment by GoogleCodeExporter** [comment]
```
I'd have to agree with Hogan's use case. That sounds what annotations would 
ideally be used for. The 'Events' section in Analytics is for on-site click 
tracking (such as, opened a modal window or played a video) while annotations 
are for factors outside of the site itself (such as a trade show or an email 
newsletter).
```

Original comment by `step...@raventools.com` on 31 Jul 2013 at 3:41

**Comment by GoogleCodeExporter** [comment]
```
A common usecase of Annotations is to highlight the events on the history of 
the site: new features where added, new version of the site, a particular bug 
was found/fixed...etc.

It would be really convenient to be able to connect the bug tracker of choice 
to your GA account.

Or, the business is running Marketing campaigns that they may be tracking with 
a calendar in a certain application. Auto adding that would make sure that we 
are not losing information.
```

Original comment by `gro...@gmail.com` on 31 Jul 2013 at 3:48

**Comment by GoogleCodeExporter** [comment]
```
Basic CRUD operations on annotations via an API isn't too much to ask. 
Annotations are useful for any number of purposes, to track when:

- marketing events occurred
- site changes/releases occurred
- outages occurred
- 3rd-party / world events affecting site behavior occurrred (e.g., getting 
Slashdotted, a holiday occurring that impacts traffic, etc)
- marking the start of tests

Annotations are timeline metadata, to be compared against any other metric. 
Annotations have no quantitative value directly associated with them aside from 
the timestamp. It exists in a different space from tally-able data like Events, 
visits, etc - and is meant to be consumed alongside with it.

The idea behind exposing the API is not so we don't have to track this stuff 
internally - but specifically so we can, and have that data sync (back and 
forth) with Google Analytics.

Another nice feature would be being able to tag these annotations for 
categorization purposes, but I'll settle for the CRUD ops.
```

Original comment by `c.filipe...@gmail.com` on 31 Jul 2013 at 6:19

**Comment by GoogleCodeExporter** [comment]
```
This woudl be great.

```

Original comment by `komal.se...@gmail.com` on 3 Sep 2013 at 8:59

**Comment by GoogleCodeExporter** [comment]
```
Please, do it. I have to enter manually more than 100 annotations manually...
```

Original comment by `torsan...@gmail.com` on 5 Sep 2013 at 9:53

**Comment by GoogleCodeExporter** [comment]
```
The feature is, I think, absolutely necessary to make the annotations function 
what it really should have been from the start
```

Original comment by `geekwith...@gmail.com` on 19 Sep 2013 at 9:14

**Comment by GoogleCodeExporter** [comment]
```
Lost any hopes.... just look for different products, there are many!!
```

Original comment by `mcons...@gmail.com` on 25 Sep 2013 at 7:59

**Comment by GoogleCodeExporter** [comment]
```
Yes please!
```

Original comment by `aidanas...@gmail.com` on 27 Sep 2013 at 8:59

**Comment by GoogleCodeExporter** [comment]
```
Doesnt seem google wants to add this. Been tracking for nearly 2 years and 
still no visible movement. :(
```

Original comment by `wizardin...@gmail.com` on 28 Sep 2013 at 9:17

**Comment by GoogleCodeExporter** [comment]
```
wizardin... My guess would be that they haven't even seen it. Also, at 89 
comments over 2 years, it's not super active. 
```

Original comment by `fil...@ancientfaces.com` on 29 Sep 2013 at 6:29

**Comment by GoogleCodeExporter** [comment]
```
please :)
```

Original comment by `gil...@leetix.com` on 30 Sep 2013 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
Yes please!
```

Original comment by `twentysi...@gmail.com` on 1 Oct 2013 at 9:49

**Comment by GoogleCodeExporter** [comment]
```
l.gudeoglu@hotmail.com
```

Original comment by `nusretnakliyat35@gmail.com` on 4 Oct 2013 at 8:01

**Comment by GoogleCodeExporter** [comment]
```
Issue 75 has been merged into this issue.
```

Original comment by `pfrise...@google.com` on 8 Oct 2013 at 8:10

**Comment by GoogleCodeExporter** [comment]
```
Please add the ability to get a users Annotations.   Use: A very useful 
dimension in my Google analytic's cube :)
```

Original comment by `laurl...@gmail.com` on 14 Nov 2013 at 8:56

**Comment by GoogleCodeExporter** [comment]
```
Bump. Either bulk annotation upload, or automated annotations triggered by blog 
posts, social activity, etc. This would be fantastic.
```

Original comment by `jonat...@thrivenetmarketing.com` on 18 Nov 2013 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
Please add this, please, it should be fairly straight forward and would make 
this SOOOO much more useful
```

Original comment by `Gwilym.H...@miniclip.com` on 11 Dec 2013 at 9:00

**Comment by GoogleCodeExporter** [comment]
```
Yes please!
```

Original comment by `yevgeny....@seotime.ru` on 11 Dec 2013 at 9:19

**Comment by GoogleCodeExporter** [comment]
```
would be great to get this feature!
```

Original comment by `wusat...@googlemail.com` on 12 Dec 2013 at 10:22

**Comment by GoogleCodeExporter** [comment]
```
I'm very interested in seeing this, so I can import events from multiple 
locations and aggregate them for reporting purposes. Since business users 
prefer to annotate directly in GA, not having easy access to these 
significantly reduces the usefulness of these reports.
```

Original comment by `nationwi...@gmail.com` on 8 Jan 2014 at 2:34

**Comment by GoogleCodeExporter** [comment]
```
I've love to be able to add annotations as part of my build pipeline. Being 
able to tie in changes to code to an affect on traffic would helpful.
```

Original comment by `demetri....@gmail.com` on 20 Jan 2014 at 4:36

**Comment by GoogleCodeExporter** [comment]
```
This will help us monitor out all the different changes in the marketing mix 
has affected Conversion rate. Is quite a task to add this manually. So the 
perfect solution would be a link with the API from a Google doc.... and the 
Google doc to be linked to an online form.
```

Original comment by `alessand...@which.co.uk` on 17 Feb 2014 at 11:37

**Comment by GoogleCodeExporter** [comment]
```
I would love to make this a feature for our platform. Hundreds of thousands of 
users would benefit from it.
```

Original comment by `JoeJon...@gmail.com` on 11 Mar 2014 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
This would be incredibly valuable. Currently we manually update all annotations 
to note when we have sent newsletters, etc. Automating this process would save 
time, increase accuracy, and provide incredibly valuable data for our clients 
and our company.
```

Original comment by `j...@ideaden.com` on 16 May 2014 at 4:43

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Certainly would find this a useful feature - especially with write access, to 
allow points in time to be generated by products integrating with Google 
Analytics, now that write access to profiles and goals is being worked on in 
whitelisted beta in the Management API. It would be great if could be 
considered for about the time when Management API write access is made 
available generally, or shortly after - I couldn't imagine it'd be a big jump 
once the big aspects of profile/goal/etc. editing is made public.

It would also fit in with the ethos of the Universal Analytics "upload 
everything into GA" mentality being promoted. I'm not sure whether read-only 
access to annotations would be as crucial, but it would still be a help in this 
case.

I'd imagine it would probably need to sit in a similar position to Content 
Experiments - as a "management.annotation" resource living underneath the 
profile, but like experiments requires the profile ID for listing; no "~all" 
access.

(With v3 of the Analytics APIs, I'd suggest that this bug is miscategorised 
after the automatic component change from the sole Data Export API to Core 
Reporting, as it'd be a Management API feature now.)
```

Original comment by `lwright...@gmail.com` on 27 Jun 2014 at 2:34

**Comment by GoogleCodeExporter** [comment]
```
I'd love to use the Analytic API to sync the Annotations with a google 
calendar.  So, I can see/share the annotations in a calendar format.  So, 
please add this to your universal analytics update.
```

Original comment by `sstaw...@screenflex.com` on 16 Jul 2014 at 2:38

**Comment by GoogleCodeExporter** [comment]
```
+1 api for creating google annotations!!
```

Original comment by `adelain...@hipagesgroup.com.au` on 30 Jul 2014 at 1:25

**Comment by GoogleCodeExporter** [comment]
```
It's strange that the whole annotation system isn't available trough an API yet.
```

Original comment by `pie...@siteoptimo.com` on 4 Aug 2014 at 11:00

**Comment by GoogleCodeExporter** [comment]
```
+1 on this to enhance the power of analytics
```

Original comment by `da...@broadplace.com` on 5 Aug 2014 at 8:56

**Comment by GoogleCodeExporter** [comment]
```
Just spoke to Justin Cutroni and asked him to prioritize it.
```

Original comment by `daniel.crompton` on 8 Aug 2014 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
+1 Would like twitter annotations to be possible, would make life easier

```

Original comment by `john.lan...@oxygenna.com` on 13 Aug 2014 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
It would be great to be able to call this proposed Annotation API from 
deployment tools like OctopusDeploy to mark deployments.
```

Original comment by `lee.engl...@gmail.com` on 29 Aug 2014 at 1:38

**Comment by GoogleCodeExporter** [comment]
```
+1 api for creating google annotations!!
```

Original comment by `jure...@gmail.com` on 3 Sep 2014 at 12:48

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
It would be great to be able to upload changes from other advertising tools 
such as when budget was changed, a campaign went live etc so our GA users can 
get a better idea of what's driving results. 
```

Original comment by `garnieri...@gmail.com` on 20 Sep 2014 at 2:00

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Could anyone add more information about this? This topic exists for four years 
now, still not any information about a planning or statusupdate (or even a 
priority).

This would be extremely helpfull for the same reasons everyone else posts, I 
also can't make a hack to do it in the meanwhile, so again, this would be 
extremely usefull.

I understand this takes time to build, but a simple 'extract items [from-till 
|| limit=10]' and a 'post item [date][text]' wouldn't be the most complicated 
piece of code (I think ;) ).
```

Original comment by `techn...@jewelsite.nl` on 7 Oct 2014 at 7:13

**Comment by GoogleCodeExporter** [comment]
```
I would also like to vote for this feature. As part of our BLOX TownNews.com 
Content Management System for media sites, it would be cool to allow actions in 
the CMS (a new ad campaign, change of design or navigation, publishing an 
important story, for example) to trigger an annotation in the analytics.
```

Original comment by `cmast...@townnews.com` on 18 Dec 2014 at 4:11

**Comment by GoogleCodeExporter** [comment]
```
this would be very helpful in proving the impact of mailings on web traffic!
```

Original comment by `beetlesa...@gmail.com` on 7 Jan 2015 at 3:33

**Comment by GoogleCodeExporter** [comment]
```
+1 for creating google annotations for deployments via api
```

Original comment by `adelain...@hipagesgroup.com.au` on 5 Feb 2015 at 2:15

**Comment by GoogleCodeExporter** [comment]
```
+1 on this, especially for GA mobile. Would be extremely useful to have an 
annotation automatically added (with the APK version number) every time I push 
a build to the Google Play Dev Console
```

Original comment by `neiljkri...@gmail.com` on 22 Feb 2015 at 3:35

**Comment by GoogleCodeExporter** [comment]
```
Vital - need to add this so when client adds articles to her website the cms 
can automatically add it as an annotation so that when she gets her emailed 
report she can see what happens when she posts. Very difficult to analyse a set 
of data without all the facts available.
```

Original comment by `simonm...@googlemail.com` on 26 Feb 2015 at 3:16

**Comment by GoogleCodeExporter** [comment]
```
This is so needed! Please add this feature to the API.
```

Original comment by `jtay...@imegre.com` on 4 Mar 2015 at 5:40

**Comment by GoogleCodeExporter** [comment]
```
plus 1
```

Original comment by `d...@holstee.com` on 5 Mar 2015 at 11:56

**Comment by GoogleCodeExporter** [comment]
```
+1 super useful for pushing commit id's, would love to see this get into the 
API.
```

Original comment by `gooogleg...@gmail.com` on 17 Mar 2015 at 7:09

**Comment by GoogleCodeExporter** [comment]
```
This is so needed! Please add this feature to the API.

```

Original comment by `raf...@rafaelfurquim.com.br` on 20 Mar 2015 at 2:28

**Comment by GoogleCodeExporter** [comment]
```
plus 1
```

Original comment by `aryrab...@gmail.com` on 26 Mar 2015 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
Would be immensely helpful.
```

Original comment by `smh...@gmail.com` on 31 Mar 2015 at 6:34

**Comment by GoogleCodeExporter** [comment]
```
I would certainly like the ability to write annotations via API:

If it was possible to create annotations via the Management API then I think 
that would open up lots of useful possibilities for setting them in a 
structured way -- for example as part of the process of sending out an email or 
launching some other form of campaign, or launching a new a/b text, and so on.

Similarly, organisations which use external tools for reporting, dashboards etc 
would benefit from being able to use annotations within those interfaces. And, 
indeed, creating annotations from within their interface.

I also wonder whether the annotations system to could be expanded to allow for 
commentary widgets within the GA dashboards.

As a starting point for the native dashboards, how about a dashboard widget for 
the existing form of annotations? That might help develop their use.
```

Original comment by `tim.lbo...@screenpages.com` on 9 Apr 2015 at 4:55

**Comment by GoogleCodeExporter** [comment]
```
+1 for this. it would be incredible if the API allowed pulling as well as 
pushing new annotations.
```

Original comment by `petr.hav...@optimics.cz` on 10 Apr 2015 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
+1 - It would be a great addition to the API
```

Original comment by `a...@untanglemyweb.com` on 15 Apr 2015 at 2:38

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Anybody knows these tool http://www.annoapi.com/? They tell that can 
automatically add or update annotations in Google Analytics via API.
```

Original comment by `rodrigo....@vivareal.com` on 12 May 2015 at 9:39

**Comment by GoogleCodeExporter** [comment]
```
This would be really useful, please add this...
```

Original comment by `alistair...@glyndebourne.com` on 1 Jul 2015 at 3:39

**Comment by GoogleCodeExporter** [comment]
```
Over 5 years since the initial request! hayibo Google! + 1 
```

Original comment by `matt.g.e...@gmail.com` on 14 Jul 2015 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
+1 +1 +1 !!!!

The minute we can write an annotation via the API we'll be integrating it with 
the CMS we use and into our deployment tools. Incredibly useful to understand 
how key content changes are affecting traffic and behaviour.
```

Original comment by `gar...@springload.co.nz` on 18 Jul 2015 at 5:00

**Comment by GoogleCodeExporter** [comment]
```
Annotations should apply across all views, so that we do not have to constantly 
add the annotation to every view.
```

Original comment by `track...@topspotims.com` on 20 Jul 2015 at 3:31

**Comment by GoogleCodeExporter** [comment]
```
It's been 5 years since this post started. We are a premium customer with 
hundreds of properties and views managed and desperately need this feature. At 
the moment, to add annotations, we need to copy and paste them to every 
profile/view we have. It's absolutely preposterous. Either add it into the 
interface or add it into the API but give us something Google.
```

Original comment by `vinny.s...@gmail.com` on 14 Aug 2015 at 3:07

**Comment by GoogleCodeExporter** [comment]
```
+1 - we'd love this as part of the Measurement Protocol.
```

Original comment by `m...@digitaloutdoors.co.uk` on 24 Aug 2015 at 11:28

**Comment by GoogleCodeExporter** [comment]
```
+1 -- we have 100+ websites in GA and need to get automated annotations into GA 
as it is a manual nightmare.
```

Original comment by `cm...@mozilla.com` on 26 Aug 2015 at 4:52

**Comment by GoogleCodeExporter** [comment]
```
+1 Why is this not implemented yet? It would be incredibly useful!
```

Original comment by `i...@tamyca.de` on 1 Sep 2015 at 7:07

**Comment by GoogleCodeExporter** [comment]
```
This is sooooo needed! Manual annotations are such a pain!
```

Original comment by `st...@alphaquad.co.uk` on 7 Sep 2015 at 9:35

**Comment by GoogleCodeExporter** [comment]
```
This is so lame, Piwik all the way.
```

Original comment by `Luisrodr...@gmail.com` on 2 Oct 2015 at 1:35

**Comment by GoogleCodeExporter** [comment]
```
We need this, crazy that it hasn't been added to the API, we export data into 
Tableau for further analysis and without annotations it's only half useful
```

Original comment by `step...@bigtripper.net` on 11 Nov 2015 at 1:09

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
+1 - This would absolutely make our life so much easier! Please do prioritise 
it as it would help a lot of people and businesses like us!
```

Original comment by `emanuel...@gmail.com` on 26 Nov 2015 at 12:14

**Comment by GoogleCodeExporter** [comment]
```
Yes please.. I assume Annotations are probably being refactored and redone- 
that this hasn't been sorted already. If not please do it.  !
```

Original comment by `gk...@inviqa.com` on 30 Nov 2015 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
annotation in the api...when do we want it...now! Please...thank you kindly!
```

Original comment by `doug...@schuh.co.uk` on 3 Dec 2015 at 3:54

**Comment by GoogleCodeExporter** [comment]
```
Really disappointed that this is not available in the API. Really reduces the 
usefulness of the annotation feature. 
```

Original comment by `dom.b...@gmail.com` on 10 Dec 2015 at 2:27

**Comment by GoogleCodeExporter** [comment]
```
We really need this feature !
please rank up 
```

Original comment by `t.chano...@of2m.fr` on 15 Dec 2015 at 9:46

**Comment by GoogleCodeExporter** [comment]
```
+1 as it would make our life much easier
```

Original comment by `csaba.sz...@skyscanner.net` on 22 Dec 2015 at 1:57

**Comment by GoogleCodeExporter** [comment]
```
This would definately make Google Analytics more useful.

We need this asap.
```

Original comment by `carpusme...@gmail.com` on 1 Jan 2016 at 5:07

**Comment by GoogleCodeExporter** [comment]
```
It is incredible that 6 years later we are still waiting for this feature. 
Particularly with the good support for APIs that Google offers in other areas. 
Here is a good post that was shared with the MeasureCamp community using the 
Measurement Protocol but it is still limited and does not place the data in the 
same location as the regular reporting: 
http://connectedwindow.com/data-and-measurement/automatic-annotations-in-google-
analytics/

Would it be possible to get an update from Google regarding this feature 
request?
```

Original comment by `captain...@gmail.com` on 11 Jan 2016 at 3:05

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
This would come in verry handy ... example to empower our infrastructure team 
to add annotations when there is a change in our hosting setup 
(changemangement)... please foresee this feature in the API. thanks
```

Original comment by `vlerickm...@gmail.com` on 5 Feb 2016 at 8:59
